src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/DefaultRazorSemanticTokenInfoServiceTest.cs

```diff
@@ -94,29 +78,18 @@ public async Task GetSemanticTokens_CSharp_Nested_HTML()
 
             var csharpTokens = new int[]
             {
-                14, 0, 3, RazorSemanticTokensLegend.CSharpKeyword, 0,
-                1, 0, 1, RazorSemanticTokensLegend.CSharpVariable, 0,
-                1, 0, 1, RazorSemanticTokensLegend.CSharpOperator, 0,
-                1, 0, 1, RazorSemanticTokensLegend.CSharpPunctuation, 0,
-                1, 0, 6, RazorSemanticTokensLegend.CSharpString, 0,
-                1, 0, 1, RazorSemanticTokensLegend.CSharpPunctuation, 0,
-                1, 0, 1, RazorSemanticTokensLegend.CSharpPunctuation, 0,
+                29, 6, 3, RazorSemanticTokensLegend.CSharpKeyword, 0, // var
```

Comment:
Ouch I forgot we had the hardcoded C# tokens.

Food for thought, no action in this PR: I wonder if we'd be able to benefit from faking C# colorization by only using a local syntactic classifier to only syntactically color C# in tests so we don't need to do all the manual labor of generating this type of information.

Reply:
Ohh good idea! I wonder if we could even expose something on the C# side via external access that would enable us to access the LSP semantic tokens logic directly for testing purposes...